xbmc/addons/VFSEntry.cpp

```diff
@@ -0,0 +1,666 @@
+{
+  if (!Initialized())
+    return false;
+
+  CVFSURLWrapper url2(url);
+  return m_pStruct->Delete(&url2.url);
+}
+
+bool CVFSEntry::Rename(const CURL& url, const CURL& url2)
+{
+  if (!Initialized())
+    return false;
+
+  CVFSURLWrapper url3(url);
+  CVFSURLWrapper url4(url2);
+  return m_pStruct->Rename(&url3.url, &url4.url);
+}
+
+void CVFSEntry::ClearOutIdle()
+{
+  if (!Initialized())
+    return;
+
+  m_pStruct->ClearOutIdle();
+}
+
+void CVFSEntry::DisconnectAll()
+{
+  if (!Initialized())
+    return;
+
+  m_pStruct->DisconnectAll();
+}
+
+bool CVFSEntry::DirectoryExists(const CURL& url)
+{
+  if (!Initialized())
+    return false;
+
+  CVFSURLWrapper url2(url);
+  return m_pStruct->DirectoryExists(&url2.url);
+}
+
+bool CVFSEntry::RemoveDirectory(const CURL& url)
+{
+  if (!Initialized())
+    return false;
+
+  CVFSURLWrapper url2(url);
+  return m_pStruct->RemoveDirectory(&url2.url);
+}
+
+bool CVFSEntry::CreateDirectory(const CURL& url)
+{
+  if (!Initialized())
+    return false;
+
+  CVFSURLWrapper url2(url);
+  return m_pStruct->CreateDirectory(&url2.url);
+}
+
+static void VFSDirEntriesToCFileItemList(int num_entries,
+                                         VFSDirEntry* entries,
+                                         CFileItemList& items)
+{
+  for (int i=0;i<num_entries;++i)
+  {
+    CFileItemPtr item(new CFileItem());
+    item->SetLabel(entries[i].label);
+    item->SetPath(entries[i].path);
+    item->m_dwSize = entries[i].size;
+    //item->m_dateTime = entries[i].mtime;
+    item->m_bIsFolder = entries[i].folder;
+    if (entries[i].title)
+      item->m_strTitle = entries[i].title;
+    for (int j=0;j<entries[i].num_props;++j)
+    {
+      if (strcasecmp(entries[i].properties[j].name, "propmisusepreformatted") == 0)
+      {
+        if (strcasecmp(entries[i].properties[j].name, "true") == 0)
+          item->SetLabelPreformated(true);
+        else
+          item->SetLabelPreformated(false);
+      } else
+        item->SetProperty(entries[i].properties[j].name,
+                          entries[i].properties[j].val);
+    }
+    items.Add(item);
+  }
+}
+
+bool CVFSEntry::GetDirectory(const CURL& url, CFileItemList& items,
+                             void* ctx)
+{
+  if (!Initialized())
+    return false;
+
+  VFSCallbacks callbacks;
+  callbacks.ctx = ctx;
+  callbacks.GetKeyboardInput = CVFSEntryIDirectoryWrapper::DoGetKeyboardInput;
+  callbacks.SetErrorDialog = CVFSEntryIDirectoryWrapper::DoSetErrorDialog;
+  callbacks.RequireAuthentication = CVFSEntryIDirectoryWrapper::DoRequireAuthentication;
+
+  VFSDirEntry* entries;
+  int num_entries;
+  CVFSURLWrapper url2(url);
+  void* ctx2 = m_pStruct->GetDirectory(&url2.url, &entries, &num_entries, &callbacks);
+  if (ctx2)
+  {
+    VFSDirEntriesToCFileItemList(num_entries, entries, items);
+    m_pStruct->FreeDirectory(ctx2);
+
+    return true;
+  }
+
+  return false;
+}
+
+bool CVFSEntry::ContainsFiles(const CURL& url, CFileItemList& items)
```

Comment:
It seems this method does a lot of different things, any chance it could be split up?